apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx

```diff
@@ -0,0 +1,258 @@
+import {
+  useAgentInstances,
+  useGetActivityChecker,
+  useGetAgentIds,
+  useGetConfigHash,
+  useGetLivenessPeriod,
+  useGetMaximumInactivityPeriods,
+  useGetMinimumStakingDeposit,
+  useGetMinimumStakingPeriods,
+  useGetMultisigThreshold,
+  useTimeForEmissions,
+} from 'hooks/useGetStakingConstants';
+import { useAppSelector } from 'store/index';
+import { MyStakingContract } from 'types/index';
+
+import {
+  ActivityCheckerAddressLabel,
+  AgentIdsLabel,
+  AgentInstancesLabel,
+  LivenessPeriodLabel,
+  MaximumInactivityPeriodsLabel,
+  MaximumStakedAgentsLabel,
+  MinimumStakingDepositLabel,
+  MinimumStakingPeriodsLabel,
+  MultisigThresholdLabel,
+  RewardsPerSecondLabel,
+  ServiceConfigHashLabel,
+  TimeForEmissionsLabel,
+} from '../FieldConfig';
+import { ColFlexContainer } from './helpers';
+import { useMaxNumServices, useRewardsPerSecond } from './hooks';
+
+const { Text } = Typography;
+
+const ShowContent = ({ isLoading, data }: { isLoading: boolean; data?: string | ReactNode }) => {
+  const { networkId } = useAppSelector((state) => state.network);
+
+  if (!networkId || isLoading) return <Skeleton.Input active size="small" />;
+  return <Text>{data || NA}</Text>;
+};
+
+const ShowNetworkAddress = ({ address }: { address: Address }) => {
+  const { networkId } = useAppSelector((state) => state.network);
+
+  if (!networkId) return null;
+  if (!address) return null;
+
+  const truncatedAddress = truncateAddress(address);
+  return (
+    <a href={`${EXPLORER_URLS[networkId]}/address/${address}`} target="_blank" rel="noreferrer">
+      {`${truncatedAddress} ${UNICODE_SYMBOLS.EXTERNAL_LINK}`}
+    </a>
+  );
+};
+
+const MaximumStakedAgents: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useMaxNumServices({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const Rewards: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useRewardsPerSecond({ address });
+  return <ShowContent isLoading={isLoading} data={`${data} OLAS` || NA} />;
+};
+
+const MinimumStakingDeposit: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useGetMinimumStakingDeposit({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const MinimumStakingPeriods: FC<{ address: Address }> = ({ address }) => {
+  const { isLoading: isLivenessPeriodLoading, data: livenessPeriod } = useGetLivenessPeriod({
+    address,
+  });
+  const { isLoading: isMinimumStakingPeriodsLoading, data: minimumStakingPeriods } =
+    useGetMinimumStakingPeriods({ address });
+
+  const isLoading = useMemo(
+    () => isLivenessPeriodLoading || isMinimumStakingPeriodsLoading,
+    [isLivenessPeriodLoading, isMinimumStakingPeriodsLoading],
+  );
+  const data = useMemo(() => {
+    if (!livenessPeriod) return NA;
+    if (!minimumStakingPeriods) return NA;
+
+    const minStakingDuration = Number(minimumStakingPeriods) / Number(livenessPeriod);
```

Comment:
I assume there's a confusion in naming here, should be `minimumStakingPeriods` = `minStakingDuration` / `livenessPeriod`

Reply:
yes, the name of the component is `MinimumStakingPeriods` but here we're calculating and returning minStakingDuration which is another variable